code/modules/overmap/contacts/contact_sensors.dm

```diff
@@ -0,0 +1,154 @@
+#define SENSORS_DISTANCE_COEFFICIENT 5
+/obj/machinery/computer/ship/sensors
+	var/list/objects_in_view = list() // Associative list of objects in view -> identification process
+	var/list/contact_datums = list()
+	var/list/trackers = list()
+
+/obj/machinery/computer/ship/sensors/Destroy()
+	objects_in_view.Cut()
+	trackers.Cut()
+
+	for(var/key in contact_datums)
+		var/datum/overmap_contact/record = contact_datums[key]
+		qdel(record)
+	contact_datums.Cut()
+	. = ..()
+
+/obj/machinery/computer/ship/sensors/attempt_hook_up(obj/effect/overmap/visitable/ship/sector)
+	. = ..()
+	if(. && linked && !contact_datums[linked])
+		var/datum/overmap_contact/record = new(src, linked)
+		contact_datums[linked] = record
+		record.marker.alpha = 255
+
+/obj/machinery/computer/ship/sensors/proc/reveal_contacts(var/mob/user)
+	if(user && user.client)
+		for(var/key in contact_datums)
+			var/datum/overmap_contact/record = contact_datums[key]
+			if(record)
+				user.client.images |= record.marker
+
+/obj/machinery/computer/ship/sensors/proc/hide_contacts(var/mob/user)
+	if(user && user.client)
+		for(var/key in contact_datums)
+			var/datum/overmap_contact/record = contact_datums[key]
+			if(record)
+				user.client.images -= record.marker
+
+/obj/machinery/computer/ship/sensors/Process()
+	..()
+	update_sound()
+	if(!linked)
+		return
+
+	// Update our 'sensor range' (ie. overmap lighting)
+	if(!sensors || !sensors.use_power || !sensors.powered())
+		linked.set_light(0)
+		// TODO move to on_power_change()?
+		for(var/key in contact_datums)
+			var/datum/overmap_contact/record = contact_datums[key]
+			if(record.effect == linked)
+				continue
+			qdel(record) // Immediately cut records if power is lost.
+
+		objects_in_view.Cut()
+		return
+
+	var/sensor_range = round(sensors.range,1)
+	linked.set_light(1, sensor_range, sensor_range+1)
```

Comment:
are theses set light things needed anymore? Since area isn't even dynamically lit

Reply:
I wasn't entirely sure why it was left in in the original PR. I don't see any reason it's necessary though so I'll cut it.